sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightPlugin.kt

```diff
@@ -47,7 +47,13 @@ abstract class SqlDelightPlugin : Plugin<Project> {
     }
 
     extension = project.extensions.create("sqldelight", SqlDelightExtension::class.java)
-    extension.project = project
+    extension.databases.all {
```

Comment:
What about adding the defaults in the `SqlDelightDatabase` class via `init` or using the `objectFactory` instead?

Reply:
And use `configureEach`?

Reply:
> What about adding the defaults in the SqlDelightDatabase class via init or using the objectFactory instead?

It seems the convention with Gradle plugins is to have the plugin manage the task configuration, instead of doing it in the task itself, so I've used that here, but either way will work.

> And use configureEach?

I know this is important for `TaskContainer` to avoid unnecessary task configuration, but for this use case I would assume it's not necessary - since every database that's added to the container is going to be accessed anyway when each task is registered.

Reply:
Regarding task configuration inside the task: I don't think so, otherwise you can't register a new task without setting all configurations by yourself, see for a sample: https://github.com/JetBrains/gradle-grammar-kit-plugin/issues/108